mlflow/entities/model_registry/prompt_version.py

```diff
@@ -40,67 +62,150 @@ class PromptVersion(_ModelRegistryEntity):
         aliases: List of aliases for this prompt version. Optional.
         last_updated_timestamp: Timestamp of last update. Optional.
         user_id: User ID that created this prompt version. Optional.
+
     """
 
     def __init__(
         self,
         name: str,
         version: int,
-        template: str,
+        template: Union[str, list[dict[str, "ContentType"]]],
         commit_message: Optional[str] = None,
         creation_timestamp: Optional[int] = None,
         tags: Optional[dict[str, str]] = None,
         aliases: Optional[list[str]] = None,
         last_updated_timestamp: Optional[int] = None,
         user_id: Optional[str] = None,
+        response_format: Optional[Union[BaseModel, dict[str, Any]]] = None,
     ):
+        from mlflow.types.chat import ChatMessage
+
         super().__init__()
 
         # Core PromptVersion attributes
         self._name: str = name
         self._version: str = str(version)  # Store as string internally
         self._creation_time: int = creation_timestamp or 0
 
-        # Store template text as a tag
+        # Initialize tags first
         tags = tags or {}
-        tags[PROMPT_TEXT_TAG_KEY] = template
+
+        # Determine prompt type and set it
+        if isinstance(template, list) and len(template) > 0:
+            try:
+                all(ChatMessage.model_validate(msg) for msg in template)
```

Comment:
hmm, why is `all` used here?

Reply:
Good catch, that was probably mistakenly used when he addressed our comments. Let's refactor as a follow up